src/controllers/authentication.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable no-unsafe-optional-chaining */
+import express from 'express'
+import User from '../database/models/userModel'
+import Token from '../database/models/tokenModel'
+
+const verifyEmail = async (req: express.Request, res: express.Response) => {
+  try {
+    const { token } = req.params
+
+    if (!token) {
+      return res.status(400).send({ message: 'Bad request' })
+    }
```

Comment:
This can be done via a middleware since you'll have many API endpoints that will need Authentication